posthog/client.py

```diff
@@ -144,6 +144,7 @@ def __init__(
         exception_autocapture_integrations=None,
         project_root=None,
         privacy_mode=False,
+        before_send=None,
```

Comment:
interestingly adding a type to this parameter kicked mypy into being strict about the other parameters and we got ~14 errors

i'm going to punt on that for now since this PR doesn't introduce them